forge-game/src/main/java/forge/game/cost/CostPartWithTrigger.java

```diff
@@ -0,0 +1,42 @@
+package forge.game.cost;
+
+import java.util.Map;
+
+import com.google.common.collect.Maps;
+
+import forge.game.card.Card;
+import forge.game.card.CardCollectionView;
+import forge.game.player.Player;
+import forge.game.spellability.SpellAbility;
+import forge.game.trigger.Trigger;
+import forge.game.trigger.TriggerHandler;
+import forge.game.trigger.TriggerType;
+
+public abstract class CostPartWithTrigger extends CostPartWithList {
+
+    public CostPartWithTrigger(final String amount, final String type, final String description) {
+        super(amount, type, description);
+    }
+
+    @Override
+    protected final void handleBeforePayment(Player ai, SpellAbility ability, CardCollectionView targetCards) {
+        if (payingTrigSA != null) {
+            Card source = targetCards.get(0);
```

Comment:
shouldn't it just be payingTrigSA.getHostCard ?

Reply:
if it's a Clone it would be wrong host